bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceAction.java

```diff
@@ -412,7 +412,7 @@ private void showDialog() {
 	}
 
 	private void showOverlayInEditor() {
-		if (overlay == null) {
+		if (overlay == null || !overlay.isOverlayOpen()) {
```

Comment:
This will result in always creating a new overlay, so all the logic in the overlay used for re-opening it will be unused, won't it? Why is this necessary?